projects/packages/sync/src/modules/class-users.php

```diff
@@ -380,7 +380,11 @@ public function authenticate_handler( $user, $username, $password ) {
 			)
 		);
 
-		return $user;
+		return new \WP_Error(
```

Comment:
@briancolinger we can't modify this as it changes the current functionality which we don't want at this time.

Please revert so it still returns $user always.

Instead we need to update our testing rules so that we validate if the flags (add_flags) were updated.

Reply:
apologies, i didn't read the function in full and was simply going off the doc comments :)

Reply:
Could you please clarify this for me?
Should I revert this change?

I thought it would be ok/consistent with the existing documented @return \WP_Error|\WP_User

Reply:
Yes, I agree that we should revert to previous behavior, unless there's a good reason we should interrupt the authentication process. Before we were returning whatever we had been passed even when a problem was found because we didn't want to interrupt the user - we just wanted to set the flags that would get synced to the cache site and possibly then warn the administrator.

Reply:
Fixed.